lib/Translation_unit.ml

```diff
@@ -297,7 +298,10 @@ let format xunit ?output_file ~input_name ~source ~parsed conf opts =
         |> List.filter_map ~f:(fun (s, f_opt) ->
                Option.map f_opt ~f:(fun f -> (s, String.sexp_of_t f)))
       in
-      match parse xunit.parse conf ~source:fmted with
+      let fmted_parsable =
+        if opts.Conf.partial then xunit.make_parsable fmted else fmted
```

Comment:
Shouldn't we do that only if the parse failed ?

Reply:
Also, this seems to ignore parsing errors. Should we show them ? (eg. as warnings)

Reply:
If this option is set, I think the parsing errors should be ignored.
The code is written this way to avoid having to duplicate too much code and have a minimal impact on the current implementation. (as this PR is already hard enough to review)

Reply:
Why would this make duplicated code or a big impact ?

Showing all the parsing errors can be unhelpful indeed. What about showing a single warning in this case ? (eg. "This file contains parsing errors and is not fully formatted")

If you are concerned about the emacs plugin opening buffers, note that we can modify it.

Reply:
I think we can assume that errors are already reported by the editor.

Reply:
Not necessarily, OCamlformat must parse the entire file so it's sensitive to all parsing errors that the editor won't see. OCamlformat may be called more often than the compiler.

Also, the editor will most probably always pass this option. The user may be confused by the "Applied OCamlformat" message but nothing changed.

Reply:
I applied the recovery only when the parsing failed and a warning is emitted (maybe more than once, should be fixed in another PR)